File: DependencyInjection/DependencyInjection/_1_ControlFreak/InMessages/StartMessage.cs
using System;
using DependencyInjection._1_ControlFreak.Interfaces;
using DependencyInjection._1_ControlFreak.Services;

namespace DependencyInjection._1_ControlFreak.InMessages
{
  class StartMessage : AcmeMessage
  {
    public void AuthorizeUsing(ActiveDirectoryBasedAuthorization authorizationRules)
    {
      Console.WriteLine("Authorizing Start with " + authorizationRules);
    }

    public void WriteTo(DataDestination dataDestination)
    {
      dataDestination.Add("Writing Start to " + dataDestination);
    }
  }
}

File: Java/dependency-injection/control-freak/src/main/java/com/github/grzesiek_galezowski/DependencyInjection/_1_ControlFreak/InMessages/StartMessage.java
package com.github.grzesiek_galezowski.DependencyInjection._1_ControlFreak.InMessages;

import com.github.grzesiek_galezowski.DependencyInjection._1_ControlFreak.Interfaces.AcmeMessage;
import com.github.grzesiek_galezowski.DependencyInjection._1_ControlFreak.Interfaces.DataDestination;
import com.github.grzesiek_galezowski.DependencyInjection._1_ControlFreak.Services.ActiveDirectoryBasedAuthorization;

public class StartMessage implements AcmeMessage {
    public void authorizeUsing(ActiveDirectoryBasedAuthorization authorizationRules) {
      System.out.println("Authorizing start with " + authorizationRules);
    }

    public void writeTo(DataDestination dataDestination) {
      dataDestination.add("Writing start to " + dataDestination);
    }
  }
